### ISSUE TITLE
More Identifier Issues

### ISSUE THREAD

**Issue by tomblind** [TypeScriptToLua/TypeScriptToLua#594]
- shorthand assignment of variable declared with lua keyword name should produce error
```ts
declare var local: any;
const foo = { local };
```
- use of lua keywords as variables that are undeclared should produce the correct error
```ts
const foo = local;
```
- use of ambient indentifiers with invalid characters should produce error
```ts
declare var $foo: any;
$foo = 1;
```

Feel free to add to this list.

**Comment by ark120202** [comment]
Exporting symbols with unsupported chars doesn't work (exported as `_____24local`, imported as `$local`):

```ts
export const local = 1;
export const $local = 1;
```
```lua
local ____exports = {}
____exports["local"] = 1
____exports._____24local = 1
return ____exports
```

```ts
import { local, $local } from "./foo";
console.log(local, $local);
```
```lua
local __TSTL_foo = require("foo")
local ____local = __TSTL_foo["local"]
local _____24local = __TSTL_foo["$local"]
print(____local, _____24local)
```

**Comment by ark120202** [comment]
> use of lua keywords as variables that are undeclared should produce the correct error

It also should error when it's used in shorthands:

```ts
// Errors in #596
const foo = { local: local };
// No error
const bar = { local };
```

It's also still a bit inconsistent with undeclared builtins:

```ts
const foo = { _G };
const bar = { _G: _G };
```

```lua
local foo = {_G = _____G}
local bar = {_G = _G}
```

**Comment by tomblind** [comment]
Keep 'em coming 😆

**Comment by ark120202** [comment]
Class `name` probably would be better to have original name instead of escaped one
